### PR TITLE
fix(styles, theme): surface exports that omit Spectrum Vars proactively

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Spectrum Vars are being deprecated, this removed their usage from the docs site and surfaces exports so that teams could opt-in to the future shape of theme files where those values are not present.

Adds an SWC specific CSS Custom Property, `--swc-scale-factor`, that allows us to mimic the scale differences between global size values via CSS math.

See "Test Case 2" below for results specific changes...

At current, the only patters that have not ended their use of Spectrum Vars are:
- ~`<sp-asset>`: a new version is pending from CSS~ done!
- `<sp-banner>`: already deprecated
- `<sp-quick-actions>`: already deprecated
- `<sp-split-button>`: already deprecated

### Questions
- should we document this? particularly the usage of `--swc-scale-factor`?
- should we warn in Dev Mode about the usage of these new exports?

## How has this been tested?
-   [ ] _Test case 1_
    1. Go [here](https://spectrum-lite--spectrum-web-components.netlify.app/)
    2. Test various pages and scales and colors and themes of the site
    3. See that it delivers just like the [current docs site](https://opensource.adobe.com/spectrum-web-components/index.html)
-   [ ] _Test case 2_
    1. Go [here](https://287e3961b9c86139ed47109d7c0b414b--spectrum-web-components.netlify.app/review/#ProgressBarStories/staticWhite.png)
    2. _The results got messed up a bit when hitting the intermediary run on so many failures, but I've kicked the process and in about 1.75 hours the following should all be true._ Note that any results with results listed under `new` are likely accidental.
    3. Review the changes specific to the removal of Spectrum Vars within the context of our VRT process (mostly in the stories themselves)
      - [ ] Classic w/ Lightest Color Stops: the removal of Spectrum Vars means there is no longer a "lightest" specific background color, so ALL of the tests in this family of tests fail. Express didn't previously have a custom "lightest" delivery.
      - **Overlay Stories**
        - [ ] `customizedClickContent`: at some point the customization in this click content was removed, this adds it back
        - [ ] `sideHoverDraggable` and `updated`: the color of the target is updated in non-"dark" color stops
      - [ ] StylesStories: colors updated in the Core Tokens versions
      - [ ] SwatchGroupStories: colors updated in the Core Tokens versions
      - **Large Stories**
        - [ ] All "Over Background" stories now leverage a calculated padding, which leads to sub-pixel values in the "large" layouts leading to slightly smaller paddings and some light text shift
        - [ ] Classic Dark LRT: `ButtonPrimaryFillPendingStories` seems erroneous, but it aligns with the same delivery in other stories, possible the previous capture was a flake?

## Types of changes
-   [x] New feature (non-breaking change which adds functionality)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

**Review by hunterloftis — DISMISSED** [review]
> should we document this? particularly the usage of --swc-scale-factor?

Definitely. The problem is the documentation should include action items for why you might want to try this (once you’ve found issues, what do you do about them?). I’m not sure what to recommend here.

> should we warn in Dev Mode about the usage of these new exports?

If you mean the new theme files, I don't think so - they're new imports, so existing users won't receive any changes by default. Users who opt-in by replacing theme files have ostensibly read the docs & are doing it on purpose.

A little naming bikeshedding: reviewing this & seeing "spectrum-lite" in several places, it sounds like a "new thing." But if I understand correctly, what this calls "spectrum-lite" is perhaps more accurately "spectrum-2-preview" or "spectrum-tokens-preview?" If so, maybe we should use one of those descriptive names vs a -lite moniker.

> the removal of Spectrum Vars means there is no longer a "lightest" specific background color, so ALL of the tests in this family of tests fail. Express didn't previously have a custom "lightest" delivery.

I don't see failures in the lightest VRTs - am I looking in the wrong place?

Random Q: why does the diff of https://287e3961b9c86139ed47109d7c0b414b--spectrum-web-components.netlify.app/review/#ButtonWhiteFillStories/iconSizeOverridden.png show “For testing purposes only” in what looks like hidden text?

Since this is an opt-in preview & including all of it essentially doubles the theme size, should we make sure it’s excluded from the `bundle` package? Based on [this](https://github.com/adobe/spectrum-web-components/blob/main/tools/bundle/src/index.ts#L65) I would guess that bundle is going to include this by default if we don't carve out an exception, but maybe not?

File: packages/button/stories/index.ts [hunk 1]
```diff
@@ -98,15 +98,14 @@ export const makeOverBackground =
...
                 ? 'rgb(181, 209, 211)'
-                : 'var(--spectrum-global-color-seafoam-600)';
+                : 'var(--spectrum-seafoam-900)';
         return html`
             <div
                 style="
                     --mod-actionbutton-static-content-color: ${color};
                     --mod-button-static-content-color: ${color};
                     background-color: ${color};
-                    color: ${color};
```
Comment (Rajdeepc) [inline packages/button/stories/index.ts:108]: is color: ${color}; removal deliberate?

File: packages/overlay/stories/overlay.stories.ts [hunk 2]
```diff
@@ -648,9 +644,9 @@ export const deepNesting = (): TemplateResult => {
                 <recursive-popover
                     tabindex=""
                     style="
-                        background-color: var(--spectrum-global-color-gray-100);
-                        color: var(--spectrum-global-color-gray-800);
-                        padding: var(--spectrum-global-dimension-size-225);
+                        background-color: var(--spectrum-gray-100);
+                        color: var(--spectrum-gray-800);
+                        padding: var(--spectrum-global-dimension-size-225, 22px);
```
Comment (Rajdeepc) [inline packages/overlay/stories/overlay.stories.ts:649]: Can we change this to leverage the `--swc-scale-factor` too?

**Review by Rajdeepc — COMMENTED** [review]
Looks great!!! A few places where `--swc-scale-factor` is missed. We would be ready to merge it in!

File: tools/theme/src/express/core-core-tokens.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
```
Comment (jnjosh) [inline tools/theme/src/express/core-core-tokens.ts:2]: Should this just be named `core-tokens.ts`, `core-core` seems maybe confusing?

File: packages/button/stories/index.ts (hunk 1 above)
Comment (Westbrook) [reply packages/button/stories/index.ts:108]: Yes. @hunterloftis caught an issue where this caused important story text content to appear "hidden" in that is displayed as the same color as the background.

**Comment by Westbrook** [comment]
@hunterloftis 

- added docs: https://spectrum-lite--spectrum-web-components.netlify.app/tools/core-tokens/
- normalized naming as `*-core-tokens.ts` or in two cases `core-tokens.ts`
- lightest failures look like: https://65f46043ec34d00c96dbc893--spectrum-web-components.netlify.app/review/
- great catch on that hidden test, that happened at some point in the past, but this was a great time to correct it
- only `Theme.ts` is re-exported from the `@spectrum-web-components/theme` entry point, so none of the actual themes themselves, these new ones or the old ones are included via the file you linked

**Review by hunterloftis — DISMISSED** [review]
Great docs! Hits a nice middle ground of concise-but-informative.

File: tools/theme/core-tokens.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+```
+
+Import all specific Core Tokens-only theme files based on the system in which you deliver your application:
+
+```
+import '@spectrum-web-components/theme/src/themes-core-tokens.js';
+
+// ...OR...
+
+import '@spectrum-web-components/theme/src/express/themes-core-tokens.js';
```
[the remark this answers is not in the document]
Comment (Westbrook) [reply tools/theme/core-tokens.md:27]: The implication here isn't that you'd move between one or the other. I was attempting to show, this is how you would include all of the themes in either of the variants. What's a more clear way to communicate that?

File: tools/theme/core-tokens.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+### Usage
+
+[![See it on NPM!](https://img.shields.io/npm/v/@spectrum-web-components/theme?style=for-the-badge)](https://www.npmjs.com/package/@spectrum-web-components/theme)
+[![How big is this package in your project?](https://img.shields.io/bundlephobia/minzip/@spectrum-web-components/theme?style=for-the-badge)](https://bundlephobia.com/result?p=@spectrum-web-components/theme)
+
+```
+yarn add @spectrum-web-components/theme
+```
+
+Import the side effectful registration of `<sp-thtme>` via:
```
[the remark this answers is not in the document]
Comment (Westbrook) [reply tools/theme/core-tokens.md:14]: Apparently Grammarly can't solve all my typing issues 🤣

File: tools/theme/core-tokens.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (hunterloftis) [reply tools/theme/core-tokens.md:27]: Ah! No, it's correct already, I just misread it as a "with core tokens" vs "without core tokens" comparison, not a "regular core tokens" vs "express core tokens."

[the remark this answers is not in the document]
Comment (Westbrook) [reply tools/theme/core-tokens.md:27]: Does this feel better?

---

The various theme files within the Spectrum system can be imported as follows:

```
import '@spectrum-web-components/theme/src/themes-core-tokens.js';
```

If you leverage the Express system, the same files in that context can be imported from:

```
import '@spectrum-web-components/theme/src/express/themes-core-tokens.js';
```

[the remark this answers is not in the document]
Comment (hunterloftis) [reply tools/theme/core-tokens.md]: Sounds great

**Pull request merged**